corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -622,6 +640,67 @@ def create_patient(requests, info, case_config):
             return get_patient_by_uuid(requests, response.json()['uuid'])
 
 
+@quickcache(['requests.domain_name', 'requests.base_url', 'identifier_type'])
+def get_identifier_source_id(requests, identifier_type):
+    """
+    Returns the ID of the identifier source to be used for generating
+    values for identifiers of the given type.
+    """
+    # The idgen module doesn't offer an API to list identifier sources.
+    # Log in using a requests Session, and scrape /module/idgen/manageIdentifierSources.list
+    # (Urgh.)
```

Comment:
This function inspects the OpenMRS system to determine what "identifier source" to use for generating an identifier value. I do not love how it has to do this. But the alternative is to change identifier configuration to include an (optional) identifier source. It makes configuration more complicated (and would require migrating existing configurations) which I'd really like to avoid.